include/nbl/builtin/hlsl/shared_memory_accessor.hlsl

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SHARED_MEMORY_ACCESSOR_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SHARED_MEMORY_ACCESSOR_INCLUDED_
+
+namespace nbl 
+{
+namespace hlsl
+{
+
+#ifdef BROADCAST_MEM
+struct BroadcastScratchProxy
+{
+	uint get(uint ix)
+	{
+		return BROADCAST_MEM[ix];
+	}
+
+	void set(uint ix, uint value)
+	{
+		BROADCAST_MEM[ix] = value;
+	}
+};
+#else
+struct BroadcastScratchProxy {};
+#endif
+
+#ifdef SHUFFLE_MEM 
+struct ShuffleScratchProxy
+{
+	uint get(uint ix)
+	{
+		return SHUFFLE_MEM[ix];
+	}
+
+	void set(uint ix, uint value)
+	{
+		SHUFFLE_MEM[ix] = value;
+	}
+};
+#else
+struct ShuffleScratchProxy
+{};
+#endif
+
+// REVIEW:  Bank conflict avoidance. It seems the offset for each index should be SUBGROUP_SIZE,
+//          not WORKGROUP_SIZE (assuming banks == SUBGROUP_SIZE)
```

Comment:
no, because you don't want your memory for the `.x` coordinate to alias with `.y` coordinate for all workgroups

Also you only care about bank conflicts within a single access op, and getting a single DWORD sized smem value for a single component is a single access